libs/libcommon/src/libcommon/queue.py

```diff
@@ -708,6 +693,7 @@ def _start_newest_job_and_cancel_others(self, job: JobDocument) -> JobDocument:
             # retry for 2 seconds
             with lock(key=job.unicity_id, owner=lock_owner, sleeps=[0.1] * RETRIES, ttl=LOCK_TTL_SECONDS):
                 # get all the pending jobs for the same unicity_id
+                # TODO: remove status__in=[Status.WAITING, Status.STARTED] filter
```

Comment:
in a followup PR?

Reply:
Yes, since we will remove the unfinished jobs, it won't be necessary to query by the available status. But we can remove the filter until all finished jobs have been removed from the queue (See migration in this PR).